### PR TITLE
Pop to api

### PR DESCRIPTION
**Pull request by georgesykes86**

### CODE REVIEW

File: lib/inbox.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+class Inbox {
+  constructor () {
+    this.messages = []
+  }
+
+  addMessages (messages) {
+    let newMessages = JSON.parse(messages)
+    let inbox = this
```
Comment (benfurber) [inline lib/inbox.js]: What's this about?

File: test/features/clientConnection.test.js [hunk 2]
```diff
@@ -1,30 +1,36 @@
...
-//     setTimeout(() => {
-//       expect(console.log.mock.calls[0][0]).toEqual('Connected to server')
-//       done()
-//     }, 100)
-//   })
-// })
+const MailClient = require('../../lib/mailClient')
+const net = require('net')
+
+describe('Feature Test', () => {
```
Comment (benfurber) [inline test/features/clientConnection.test.js]: Please explain what the feature test actually covers.

File: test/features/clientConnection.test.js [hunk 3]
```diff
@@ -1,30 +1,36 @@
...
+  let client
+  let server
+
+  beforeAll(async (done) => {
+    server = net.createServer()
+    server.listen({
+      host: 'localhost',
+      port: 1337
+    }, () => {
+      client = new MailClient(1337, '127.0.0.1', 5001, '127.0.0.1')
```
Comment (charmalt) [inline test/features/clientConnection.test.js]: Please explain this!

File: test/inbox.test.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  let inbox
+  const messagesString = '[{"id": 1, "mailfrom": "George", "mailto": "Charlene", "mailbody": "HI"},' +
+    '{"id": 2, "mailfrom": "John", "mailto": "Igor", "mailbody": "BYE"}]'
+  const messages = [{"id": 1, "mailfrom": "George", "mailto": "Charlene", "mailbody": "HI"},
+    {"id": 2, "mailfrom": "John", "mailto": "Igor", "mailbody": "BYE"}]
+  const moreMessagesString = '[{"id": 2, "mailfrom": "John", "mailto": "Igor", "mailbody": "BYE"},' +
+    '{"id": 3, "mailfrom": "Tom", "mailto": "Will", "mailbody": "Cheerio "}]'
+  const moreMessages = [{"id": 1, "mailfrom": "George", "mailto": "Charlene", "mailbody": "HI"},
+    {"id": 2, "mailfrom": "John", "mailto": "Igor", "mailbody": "BYE"},
+    {"id": 3, "mailfrom": "Tom", "mailto": "Will", "mailbody": "Cheerio "}]
```
Comment (benfurber) [inline test/inbox.test.js]: You're aware there's only 2 items in moreMessagesString but 3 in moreMessages, yeah?

**Pull request merged**